packages/commonwealth/client/scripts/views/components/component_kit/cw_wallets_list.tsx

```diff
@@ -207,13 +208,15 @@ export class CWWalletsList extends ClassComponent<WalletsListAttrs> {
       }
 
       try {
-        const validationBlockInfo =
-          wallet.getRecentBlock && (await wallet.getRecentBlock());
+        const sessionPublicAddress = await app.sessions.getOrCreateAddress(wallet.chain, await wallet.getChainId());
+        const chainIdentifier = app.chain?.id || wallet.defaultNetwork;
+        const validationBlockInfo = wallet.getRecentBlock && await wallet.getRecentBlock(chainIdentifier);
```

Comment:
A little concerned about this -- when does it trigger + how much overhead is it?

Reply:
This is triggered when the user clicks one of the wallet options on the login modal.

```
        const sessionPublicAddress = await app.sessions.getOrCreateAddress(wallet.chain, await wallet.getChainId());
```
This line looks up the session address from the `app.sessions` state object, the only overhead here is that it will generate a random public/private key pair if it does not exist for the given chain + chain id.

```
        const validationBlockInfo = wallet.getRecentBlock && await wallet.getRecentBlock(chainIdentifier);
```
This calls the `getRecentBlock` method on the wallet, which involves an RPC call to the selected chain. Depending on the particular RPC, this might be slow / rate limited. I talked to @raykyri yesterday about this and we came to the conclusion that we should prioritise supporting `getRecentBlock` for Ethereum and Cosmos - so we can remove the functions for Solana, Polkadot, Near etc.